test/unit/storage/patch/ImmutableMetadataPatcher.test.ts

```diff
@@ -0,0 +1,169 @@
+import { DataFactory, Store } from 'n3';
+import type { Patch } from '../../../../src/http/representation/Patch';
+import { RepresentationMetadata } from '../../../../src/http/representation/RepresentationMetadata';
+import { ImmutableMetadataPatcher } from '../../../../src/storage/patch/ImmutableMetadataPatcher';
+import type { RdfStorePatcherInput } from '../../../../src/storage/patch/RdfStorePatcher';
+import type { SparqlUpdatePatcher } from '../../../../src/storage/patch/SparqlUpdatePatcher';
+import { ConflictHttpError } from '../../../../src/util/errors/ConflictHttpError';
+import { NotImplementedHttpError } from '../../../../src/util/errors/NotImplementedHttpError';
+import { FilterPattern } from '../../../../src/util/QuadUtil';
+import { guardedStreamFrom } from '../../../../src/util/StreamUtil';
+import { PIM, RDF } from '../../../../src/util/Vocabularies';
+import { SimpleSuffixStrategy } from '../../../util/SimpleSuffixStrategy';
+import 'jest-rdf';
+const { namedNode, quad } = DataFactory;
+
+function getPatch(): Patch {
+  const prefixedQuery = `This is a valid patch query`;
+  return {
+    data: guardedStreamFrom(prefixedQuery),
+    metadata: new RepresentationMetadata(),
+    binary: true,
+    isEmpty: false,
+  };
+}
+
+describe('A ImmutableMetadataPatcher', (): void => {
+  const base = 'http://test.com/';
+  const identifier = { path: 'http://test.com/foo' };
+  const metaIdentifier = { path: 'http://test.com/foo.meta' };
+
+  let patch: Patch;
+  let patcher: SparqlUpdatePatcher;
+  let handler: ImmutableMetadataPatcher;
+  let metaStrategy: SimpleSuffixStrategy;
+  let store: Store;
+  let input: RdfStorePatcherInput;
+
+  beforeEach(async(): Promise<void> => {
+    patcher = {
+      canHandle: jest.fn(),
+      handle: jest.fn(async(): Promise<Store> => new Store([
+        quad(namedNode(`${base}s`), namedNode(`${base}p`), namedNode(`${base}o`)) ])),
+    } as unknown as SparqlUpdatePatcher;
```

Comment:
```suggestion
    } as any;
```
Typing will be known because of the variable definitions above.